### ISSUE TITLE
I start getting port error already in use

### ISSUE THREAD

**Issue by asmgx** [automl/auto-sklearn#1412]
## Describe the bug ##
I am running Ubuntu on Windows 11 bash and auto sklearn version **0.14.6**
Whenever I try to call Autosklearn I get this error
      
      **_An error ocurred while starting the kernel
      /home/asmgx/.local/lib/python3.8/site‑packages/distributed/node.py:180: UserWarning: Port 8787 is already in use.
      Perhaps you already have a cluster running?
      Hosting the HTTP server on port 42913 instead
      warnings.warn(_**

here is the code

      
      **automl = autosklearn.classification.AutoSklearnClassifier(
      ime_left_for_this_task=60*10,
      per_run_time_limit=60*1,
      memory_limit = 1024 * 10,
      n_jobs=-1,
      metric=autosklearn.metrics.f1_macro,
             )**
      

i tried to restart my laptop and restart kernel but nothing works, always getting the same error

I tried to call the code from Spyder (got same error) 
and tried from python and still same error

**Comment by JustinDoIt** [comment]
It seems to be just a warning, not an error. In other words, the program does not stop computing in the background. 

you can launch the `logging` and see the program's processing state

```python 
logging_config = {
        'version': 1,
        'disable_existing_loggers': False,
        'formatters': {
            'custom': {
                # More format options are available in the official
                # `documentation <https://docs.python.org/3/howto/logging-cookbook.html>`_
                'format': '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
            }
        },

        # Any INFO level msg will be printed to the console
        'handlers': {
            'console': {
                'level': 'INFO',
                'formatter': 'custom',
                'class': 'logging.StreamHandler',
                'stream': 'ext://sys.stdout',
            },
        },

        'loggers': {
            '': {  # root logger
                'level': 'DEBUG',
            },
            'Client-EnsembleBuilder': {
                'level': 'DEBUG',
                'handlers': ['console'],
            },
        },
    }
runtime = 3600
per_runtime = min(runtime // 10, 1800)
    automl = autosklearn.classification.AutoSklearnClassifier(
        time_left_for_this_task=runtime,
        per_run_time_limit=per_runtime,
        initial_configurations_via_metalearning=25,
        ensemble_size=50,
        ensemble_nbest=50,
        max_models_on_disc=None,
        memory_limit=None,
        tmp_folder='./tmp_folder' + str(random.randrange(1,1000)),
        delete_tmp_folder_after_terminate=True,
        n_jobs=-1,
        seed=42,
        logging_config=logging_config,
    )

```

**Comment by asmgx** [comment]
The process get Killed and does not continue

**Comment by JustinDoIt** [comment]
Hi @asmgx 
I encountered this situation repeatedly and can be sure that the program did not terminate on my machine (Ubuntu 20.04 and Ubuntu 18.04).

There will be any other error messages when the program terminates. Can you provide some logs or output information?

**Comment by eddiebergman** [comment]
This can be caused when the logging server was not shut down properly, often when auto-sklearn crashes for other reasons, eventually the logging server will shutdown once inactive but it will take a while.

When starting a new process of auto-sklearn, it will try to use the same default port, which is now occupied by this original logging server. A warning is shown that the port is in use, and another port is then assigned to it. This happens quite often in our tests where we are testing for a crash to occur in certain circumstances. 

It should not cause any issues with auto-sklearn's running though and it is just a warning which can be ignored.

I'm going to close as it's not a bug and it's not something we can really control, if a crash happens, then we can't auto-magically close the logging server. The only thing we could do is hide the warning and I don't think that is a good idea either.

Please feel free to re-open if this warning comes up in situations other than described or something was not clear!
